GetStream/stream-chat-android
# [CAS-1223] Improve push notifications docs and KDOCs

### 🎯 Goal

Improve push notifications docs and KDOCs

### 🛠 Implementation details

- Added a section about overriding icon and test resources to Docusaurus
- Added KDOCs to push notifications' related classes and methods


### 🧪 Testing
- Run Docusaurus locally
- Check KDOCs

### ☑️ Checklist
